crates/task-impls/src/consensus.rs

```diff
@@ -359,7 +360,7 @@ where
 }
 
 impl<
-        TYPES: NodeType,
+        TYPES: NodeType<BlockHeader = VIDBlockHeader, BlockPayload = VIDBlockPayload>,
```

Comment:
Same here, `BlockHeader` should be generic, possibly bounded by a `Header` trait